src/metabase/api/table.clj

```diff
@@ -442,14 +443,18 @@
                                                                         [:= :most_recent true]]
                                                              :order-by [[:id :desc]]
                                                              :limit    1}
-                                                            :id id)
+                                             :id id)
                                            first :status)
-        db (t2/select-one Database :id database_id)]
+        db (t2/select-one Database :id database_id)
+        ;; a native model can have columns with keys as semantic types on if a user specified it
+        trust-semantic-keys? (and (= (:type card) :model)
```

Comment:
A regular (mbql) model can also have metadata overrides with FKs for columns that are not FKs in the DB